mathosphere-core/src/main/java/com/formulasearchengine/mathosphere/mathpd/text/BasicStringMatcher.java

```diff
@@ -0,0 +1,189 @@
+        return reconcileOverlappings(compare(text1, text2));
+    }
+
+    /**
+     * Are two matches overlapping in both documents?
+     * <p>
+     * I use a simplification since I know the matches are ordered.
+     * This means that all positions in match2 are after match1
+     * or at the same position.
+     *
+     * @param m1 previous match
+     * @param m2 succeeding match
+     * @return do they overlap? [..[__]_]
+     */
+    boolean isOverlapping(int[] m1, int[] m2) {
+        if (m1[0] <= m2[0] && m2[0] <= m1[1]) { // [..[__] in docA
+            return m1[2] <= m2[2] && m2[2] <= m1[3]; // [..[__] in docB
+        }
+        return false;
+    }
+
+    private String normalizeString(String text1) {
+        text1 = text1.toLowerCase().replaceAll("([^a-z]+)", " ").toLowerCase();
+        return text1;
+    }
+
+    /**
+     * Mashes overlapping matches and saves every unique match into the textPattern.
+     *
+     * @param tmpMatches Ordered list of overlapping matches.
+     */
+    List<int[]> reconcileOverlappings(final List<int[]> tmpMatches) {
+        final List<int[]> resultMatches = new ArrayList<>();
+        if (tmpMatches.size() > 1) {
+            int[] curMatch = tmpMatches.get(0);
+            for (int i = 1; i < tmpMatches.size() - 1; i++) {
+                final int[] nextMatch = tmpMatches.get(i);
+                if (isOverlapping(curMatch, nextMatch)) {
+                    // expand and mash up
+                    curMatch = new int[] {
+                            Math.min(curMatch[0], nextMatch[0]),
+                            Math.max(curMatch[1], nextMatch[1]),
+                            Math.min(curMatch[2], nextMatch[2]),
+                            Math.max(curMatch[3], nextMatch[3]),
+                            Math.max(curMatch[1], nextMatch[1]) - Math.min(curMatch[0], nextMatch[0])
+                    };
+                } else {
+                    // save as a unique match and move on
+                    resultMatches.add(curMatch);
+                    curMatch = nextMatch;
+                }
+            }
+            // save the last match
+            resultMatches.add(curMatch);
+        }
+        return resultMatches;
+    }
+
+    /**
+     * Compares two texts and returns a score that represents its similarity.
+     *
+     * @param text1 Text 1
+     * @param text2 Text 2
+     * @return ordered list of matches between text1 and text2
+     * @throws Exception I would assume arrayoutofbound.
+     *                   The alphabet in BoyerMoore is too small?
+     */
+    double scoreSimilarity(String text1, String text2) throws Exception {
+
+        final java.util.List<int[]> matches = compare(text1, text2);
+        int sumMatchLength = 0;
+        for (int[] match : matches) {
+            sumMatchLength += match[4];
+        }
+        return Math.min((double) sumMatchLength / text1.length(), 1.0);
```

Comment:
I don't understand the idea of the dividing by text1.length()  (it was source.length in https://bitbucket.org/sciplore/citeplag-dev-backend/src/f7823252ccfd1780d24f03afddd986c35142d439/pds-backend-core/src/main/java/org/sciplore/pds/algorithm/impl/BasicStringMatching.java?at=systemupdate&fileviewer=file-view-default#BasicStringMatching.java-116)
And why did it work before without the (double) prefix if you divide two int values?